File: lib/regexes.rb
# Cosy home for the eight zillion command regexes
class Woolf
  module Regexes
    MATCH_ONE = '\s([a-zA-Z]*\b+)'.freeze
    MATCH_ALL = '\s(.*)'.freeze

    SPRINT = /!sprint in (\d+) for (\d+)/ unless const_defined?(:SPRINT)
    SYN = Regexp.new("!synonym#{MATCH_ONE}") unless const_defined?(:SYN)
    ANT = Regexp.new("!antonym#{MATCH_ONE}") unless const_defined?(:ANT)
    RHYME = Regexp.new("!rhyme#{MATCH_ONE}") unless const_defined?(:RHYME)
    MEANSLIKE = Regexp.new("!wordslike#{MATCH_ALL}") unless const_defined?(:MEANSLIKE)
    TRIGGER = Regexp.new("!related#{MATCH_ONE}") unless const_defined?(:TRIGGER)
    DESCRIBE = Regexp.new("!describe#{MATCH_ONE}") unless const_defined?(:DESCRIBE)
  end
end

File: src/utils/regexes.ts
const MATCH_ONE = '\\s([a-zA-Z]*\\b)';

export const COMMAND = /![a-zA-Z]/;

export const INFO = /(!\w+.+){2,}/m;

export const QUOTE = /(`|"|')!\w+/;

export const SPRINT = /!sprint in (\d+) for (\d+)/;

export const INSPIRE = /!inspiration/;

export const SUPPORT = /!woolf support/;

export const SYN = new RegExp(`!synonym${MATCH_ONE}`);

export const ANT = new RegExp(`!antonym${MATCH_ONE}`);

export const RHYME = new RegExp(`!rhyme${MATCH_ONE}`);

export const TRIGGER = new RegExp(`!related${MATCH_ONE}`);

export const DESCRIBE = new RegExp(`!describe${MATCH_ONE}`);
